compiler/sigmatch.nim

```diff
@@ -1341,12 +1375,17 @@ proc typeRel(c: var TCandidate, f, aOrig: PType,
   of tyTuple:
     if a.kind == tyTuple: result = recordRel(c, f, a)
   of tyObject:
-    if a.kind == tyObject:
-      if sameObjectTypes(f, a):
-        result = isEqual
+    let effectiveArgType = if a.kind == tyAlias: a else: getObjectTypeOrNil(a)
```

Comment:
Why does `tyAlias` remain a `tyAlias`?! The rest of the logic does not consider `tyAlias` and aliases to object types are allowed and common.

Reply:
I should have taken better notes, but I believe I tracked this down to an issue with bare type variables. You are right, this is not correct. I'll draft until I fix it.

Reply:
To follow up on this, the reason checking for `tyAlias` was done here was to "keep concepts working" as I said in one of my comments that I forgot about. When a concept is evaluating matching, particularly with the `var` modifier, it is assumed that on to `typeRel`: f:`tyObject`, aOrig: `tyTypedesc` (`typedesc[var Alias]`) should return `isNone`. This means that in order for this functionality to work in the current version of Nim, the condition for `tyAlias` never matching with `tyObject` is acceptable (assuming there is not other legal way to get that type structure). Since this doesn't really make sense, as you pointed out, I switched the logic to try and explicitly detect concept modifier matching by looking checking for `useTypeLoweringRuleInTypeClass` instead. This is essentially the same approach as before in that it just falls back on previous behavior for this case but it may read easier this way.